examples/pytorch_lightning_simple.py

```diff
@@ -37,102 +38,106 @@
         data, target = batch
-        output = self.forward(data)
+        output = self(data)
         return F.nll_loss(output, target)
 
-    def validation_step(self, batch, batch_nb):
+    def validation_step(self, batch, batch_idx: int) -> None:
         data, target = batch
-        output = self.forward(data)
+        output = self(data)
         pred = output.argmax(dim=1, keepdim=True)
         accuracy = pred.eq(target.view_as(pred)).float().mean()
         self.log("val_acc", accuracy)
 
-    def configure_optimizers(self):
-        return Adam(self.model.parameters())
+    def configure_optimizers(self) -> optim.Optimizer:
+        return optim.Adam(self.model.parameters())
+
+
+class MNISTDataModule(pl.LightningDataModule):
+    def __init__(self, data_dir: str, batch_size: int):
+        super().__init__()
+        self.data_dir = data_dir
+        self.batch_size = batch_size
 
-    def train_dataloader(self):
-        return torch.utils.data.DataLoader(
-            datasets.MNIST(DIR, train=True, download=True, transform=transforms.ToTensor()),
-            batch_size=BATCHSIZE,
-            shuffle=True,
+    def setup(self, stage: Optional[str] = None) -> None:
+        self.mnist_test = datasets.MNIST(
+            self.data_dir, train=False, download=True, transform=transforms.ToTensor()
         )
+        mnist_full = datasets.MNIST(
+            self.data_dir, train=True, download=True, transform=transforms.ToTensor()
+        )
+        self.mnist_train, self.mnist_val = random_split(mnist_full, [55000, 5000])
 
-    def val_dataloader(self):
-        return torch.utils.data.DataLoader(
-            datasets.MNIST(DIR, train=False, download=True, transform=transforms.ToTensor()),
-            batch_size=BATCHSIZE,
-            shuffle=False,
+    def train_dataloader(self) -> DataLoader:
+        return DataLoader(
+            self.mnist_train, batch_size=self.batch_size, shuffle=True, pin_memory=True
         )
 
+    def val_dataloader(self) -> DataLoader:
+        return DataLoader(
+            self.mnist_val, batch_size=self.batch_size, shuffle=False, pin_memory=True
+        )
 
-def objective(trial):
-    # Filenames for each trial must be made unique in order to access each checkpoint.
-    checkpoint_callback = pl.callbacks.ModelCheckpoint(
-        os.path.join(MODEL_DIR, "trial_{}".format(trial.number), "{epoch}"), monitor="val_acc"
-    )
+    def test_dataloader(self) -> DataLoader:
+        return DataLoader(
+            self.mnist_test, batch_size=self.batch_size, shuffle=False, pin_memory=True
+        )
+
+
+def objective(trial: optuna.trial.Trial) -> float:
+
+    model = LightningNet(trial)
+    datamodule = MNISTDataModule(data_dir=DIR, batch_size=BATCHSIZE)
 
     trainer = pl.Trainer(
         logger=False,
         limit_val_batches=PERCENT_VALID_EXAMPLES,
-        checkpoint_callback=checkpoint_callback,
+        checkpoint_callback=False,
```

Comment:
According to the [documentation](https://pytorch-lightning.readthedocs.io/en/stable/trainer.html#trainer-class-api), the method of passing a `ModelCheckpoint` instance to `checkpoint_callback` is deprecated, so passing a bool value seems to be a good idea. However, is it ok to use False here? Optuna creates a `pl.Trainer` for each trial, so if a user wants to use a checkpoint, it needs to be saved for each trial. To let users know this, shouldn't this value be set to True so that a different checkpoint is saved for each trial?

Reply:
Hi, sorry for the late response.

I think is might be not such good but i used the same logic as in previous version.

```diff
- shutil.rmtree(MODEL_DIR)
```
https://github.com/optuna/optuna/pull/2332/files#diff-318c96a1dc252cea529ff2239e3001338de49f9f54e2a73193507c45aaa9b317L167

Reply:
I thought it seemed like a really good idea to save the logs. I'll make a commit now.

Reply:
Also I have enabled hyperparameter logging
<img width="954" alt="Screenshot 2021-02-18 at 23 16 13" src="https://user-images.githubusercontent.com/7549587/108416644-0b6b8000-7240-11eb-93d1-cb5ba08b40fa.png">

Reply:
Thanks for your response. I agree with you. Let me resolve this conversation.